dash/development/_py_components_generation.py

```diff
@@ -422,25 +426,36 @@ def create_prop_docstring(prop_name, type_object, required, description,
         type_object=type_object,
         is_flow_type=is_flow_type,
         indent_num=indent_num + 1)
-
     indent_spacing = '  ' * indent_num
+
+    is_required = 'optional'
+    if required:
+        is_required = 'required'
+    elif default and 'null' not in str(default) and '[]' not in str(default):
```

Comment:
maybe also exclude `{}`?

Scares me a little to use `not in` here - like, what if we have a nested prop with a default like `{"mode": "local", "extras": []}` - can we just do `!=`?

Reply:
That makes sense! Fixed in c70659e.